lib/channel_model.js

```diff
@@ -286,6 +282,7 @@ class ConfirmChannel extends Channel {
             if (err === null) resolve();
             else reject(err);
           };
+          if (!this.pending) unconfirmed[index](new Error('channel closed'))
```

Comment:
Where is this coming from?

Reply:
This does the same as [this](https://github.com/amqp-node/amqplib/blob/aa46d7681a5811a8dcf99ee9a711f03d0a1e7f29/lib/channel.js#L38) and was necessary for a [test](https://github.com/amqp-node/amqplib/blob/aa46d7681a5811a8dcf99ee9a711f03d0a1e7f29/test/channel_api.js#L583). 'close' is emitted before `waitForConfirms` is called so it wasn't being handled there. `this.pending` is set to null when the channel is closed (in `Channel._rejectPending`) so that's used to check.

Reply:
this sounds like it could result in memory leaks. when do we dispose of `unconfirmed`?

Reply:
True, I guess. Ostensibly it's disposed in the 'close' handler in `lib/channel.js`. However, due to the check in the while loop it exits midway (`unconfirmed` seems to have values of `false`). I've updated the code to match what's in the 'close' handler.